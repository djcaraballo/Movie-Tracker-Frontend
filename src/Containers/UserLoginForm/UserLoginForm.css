.form-wrapper {
  height: 100vh;
  background: linear-gradient(rgb(0, 0, 0), rgba(131, 130, 130, 0.8), rgba(204, 96, 96, 0.3)) , url(../../Images/photo-1517604931442-7e0c8ed2963c.jpeg);
  background-size: cover;
}

.form-input {
  height: 1.5em;
  margin-bottom: 1em;
  font-size: 1.5em;
  font-family: 'Lato', sans-serif;
}

input {
  background: none;
  border: none;
  border-bottom: 2px solid #0496a0;
}

input::placeholder {
  color: #170A1C;
  text-align: left;
}


.user-login-form {
  align-items: center;
  display: flex;
  flex-direction: column;
  height: 550px;
  justify-content: space-around;
  padding-bottom: 50px;
  flex-direction: column;
  width: 70%;
  margin: 4rem auto;
  font-family: 'Lato', sans-serif;
  background: rgba(255, 255, 255, 0.6);
  border-radius: 2%;

}



.user-email-login,
.user-password-login,
.login-button {
  font-size: 1.5em;
}

.user-email-login {
  margin-top: 30px;
}

.user-email-login,
.user-password-login {
  height: 40px;
  padding-bottom: 10px;
  padding-left: 20px;
  padding-top: 10px;
  width: 30rem;
  width: 50%;
  transition: all .2s;
}

.user-email-login:focus,
.user-password-login:focus {
  outline-color: #0496a0;
}

.login-button {
  background: none;
  height: 58px;
  margin-bottom: 30px;
  width: 20rem;
  border: 2px solid #0496a0;
}

.user-signup-link {
  font-size: 1.5em;
}

@media (max-width: 600px) {
  .user-email-login,
  .user-password-login {
    font-size: 1em;
    height: 30px;
    padding-bottom: 5px;
    padding-left: 10px;
    padding-top: 5px;
    width: 85%;
  }

  .login-button {
    font-size: 1em;
    height: 48px;
    width: 86.75%;
  }

  .user-login-form {
    margin: 3rem auto;
    width: 85%;
  }
}

@media (max-width: 475px) {
  .user-login-form {
    margin: 2rem auto;
  }
}

@media (max-width: 350px) {
  .user-login-form {
    margin: 1rem auto;
  }
  width: 51.75%;
  border: 2px solid #0496a0;
  color: #170A1C;
  transition: all .5s;
}

.login-button:hover {
  background: #228CDB;
  color: white;
  transition: all .5s ease-in-out;
  transform: scale(1.1); 
}

.user-signup-link {
  border: 2px solid #0496a0;
  padding: 4% 10%;
  transition: all .5s;
  text-decoration: none;
  font-size: 1.1rem;
  color: #170A1C;
}

.user-signup-link:hover {
  background: #C19AB7;
  color: white;
  transition: all .5s ease-in-out;
  transform: scale(1.1); 
}