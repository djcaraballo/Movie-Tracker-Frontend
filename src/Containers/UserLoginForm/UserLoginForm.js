import React, { Component } from 'react';
import { signInUser } from '../../Actions/user-actions';
import { connect } from 'react-redux';
import * as API from '../../Utils/API/';

class UserLoginForm extends Component {
  constructor(props) {
    super(props)
    this.state = {
      email: '',
      password: ''
    }
  }

  handleChange = (e) => {
    const { name, value } = e.target
    this.setState({ [name]: value })
  }

  handleSubmit = async (e) => {
    e.preventDefault()
      const fetchedUser = await API.checkUser(this.state)
      this.props.userSignIn(fetchedUser) 
  }

  render() {
    const { email, value } = this.state
    return (
      <form onSubmit={this.handleSubmit}>
        <input
          className="user-email-login" 
          name="email"
          value={email}
          onChange={this.handleChange}
          placeholder="Enter Email Address" />
        <input 
          className="user-password-login"
          type="password"
          name="password"
          value={value}
          onChange={this.handleChange}
          placeholder="Enter Password" />
        <button>Go To Movies!</button>
      </form>
    )
  }
}

const mapStateToProps = (state) => ({
  user: state.user
})

const mapDispatchToProps = (dispatch) => ({
  userSignIn: (user) => {
    dispatch(signInUser(user))
  }
})

export default connect(mapStateToProps, mapDispatchToProps)(UserLoginForm)