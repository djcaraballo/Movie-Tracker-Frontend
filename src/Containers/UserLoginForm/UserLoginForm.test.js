import React from 'react';
import ReactDOM from 'react-dom';
import { UserLoginForm } from './UserLoginForm';
import { shallow } from 'enzyme';


describe('UserLoginForm', () => {
  let wrapper;
  const mockUserSignIn = jest.fn()
  const mockUser = {
    name: 'Louisa',
    id: 1806,
    email: 'Louisa@turing.io',
    password: 'oooooooooooookay'
  }

  beforeEach(() => {
    wrapper = shallow(<UserLoginForm userSignIn={mockUserSignIn} user={mockUser} />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have the correct default state', () => {
    expect(wrapper.state().email).toEqual('')
    expect(wrapper.state().password).toEqual('')
  })

  describe('handleChange function', () => {
    it('should set state with the correct keys and values', () => {

    })
  })

  describe('handleSubmit function', () => {
    it('should call checkUser with the correct params', () => {

    })

    it('should call the userSignIn action creator with the correct params', () => {

    })

    it('should throw an error if the fetch call fails', () => {

    })
  })

  describe('mapStateToProps function', () => {

  })

  describe('mapDispatchToProps function', () => {

  })
});
