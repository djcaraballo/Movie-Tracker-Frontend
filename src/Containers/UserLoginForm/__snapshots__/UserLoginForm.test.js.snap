// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserLoginForm should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserLoginForm
    user={
      Object {
        "email": "Louisa@turing.io",
        "id": 1806,
        "name": "Louisa",
        "password": "oooooooooooookay",
      }
    }
    userSignIn={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        className="user-login-form"
        onSubmit={[Function]}
      >
        <input
          className="form-input user-email-login"
          name="email"
          onChange={[Function]}
          placeholder="Enter Email Address"
          value=""
        />
        <input
          className="form-input user-password-login"
          name="password"
          onChange={[Function]}
          placeholder="Enter Password"
          type="password"
        />
        <button
          className="login-button"
        >
          Go To Movies!
        </button>
        <a
          className="user-signup-link"
          href="/newuser"
        >
          New User Sign Up
        </a>
      </form>,
      "className": "form-wrapper",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="form-input user-email-login"
            name="email"
            onChange={[Function]}
            placeholder="Enter Email Address"
            value=""
          />,
          <input
            className="form-input user-password-login"
            name="password"
            onChange={[Function]}
            placeholder="Enter Password"
            type="password"
          />,
          <button
            className="login-button"
          >
            Go To Movies!
          </button>,
          <a
            className="user-signup-link"
            href="/newuser"
          >
            New User Sign Up
          </a>,
        ],
        "className": "user-login-form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "form-input user-email-login",
            "name": "email",
            "onChange": [Function],
            "placeholder": "Enter Email Address",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "form-input user-password-login",
            "name": "password",
            "onChange": [Function],
            "placeholder": "Enter Password",
            "type": "password",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Go To Movies!",
            "className": "login-button",
          },
          "ref": null,
          "rendered": "Go To Movies!",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "New User Sign Up",
            "className": "user-signup-link",
            "href": "/newuser",
          },
          "ref": null,
          "rendered": "New User Sign Up",
          "type": "a",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          className="user-login-form"
          onSubmit={[Function]}
        >
          <input
            className="form-input user-email-login"
            name="email"
            onChange={[Function]}
            placeholder="Enter Email Address"
            value=""
          />
          <input
            className="form-input user-password-login"
            name="password"
            onChange={[Function]}
            placeholder="Enter Password"
            type="password"
          />
          <button
            className="login-button"
          >
            Go To Movies!
          </button>
          <a
            className="user-signup-link"
            href="/newuser"
          >
            New User Sign Up
          </a>
        </form>,
        "className": "form-wrapper",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className="form-input user-email-login"
              name="email"
              onChange={[Function]}
              placeholder="Enter Email Address"
              value=""
            />,
            <input
              className="form-input user-password-login"
              name="password"
              onChange={[Function]}
              placeholder="Enter Password"
              type="password"
            />,
            <button
              className="login-button"
            >
              Go To Movies!
            </button>,
            <a
              className="user-signup-link"
              href="/newuser"
            >
              New User Sign Up
            </a>,
          ],
          "className": "user-login-form",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "form-input user-email-login",
              "name": "email",
              "onChange": [Function],
              "placeholder": "Enter Email Address",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "form-input user-password-login",
              "name": "password",
              "onChange": [Function],
              "placeholder": "Enter Password",
              "type": "password",
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Go To Movies!",
              "className": "login-button",
            },
            "ref": null,
            "rendered": "Go To Movies!",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "New User Sign Up",
              "className": "user-signup-link",
              "href": "/newuser",
            },
            "ref": null,
            "rendered": "New User Sign Up",
            "type": "a",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
